cluster_pack/uploader.py

```diff
@@ -309,11 +309,25 @@ def _upload_env_from_venv(
         if not resolved_fs.exists(dir):
             resolved_fs.mkdir(dir)
         _logger.info(f'Uploading env at {local_package_path} to {package_path}')
-        resolved_fs.put(local_package_path, package_path)
+        _put(resolved_fs, local_package_path, package_path)
 
         _dump_archive_metadata(package_path, reqs, resolved_fs)
 
 
+def _put(resolved_fs: filesystem.EnhancedFileSystem,

```

Comment:
Maybe we could rename this function and even put it public as "put_safe_atomic", WDYT ?